.github/workflows/check-precommit.yml

```diff
@@ -0,0 +1,57 @@
+name: Check formatting flow
+
+on:
+  workflow_call:
+    inputs:
+      python-version:
+        description: "Python version to use"
+        default: "3.9"
+        required: false
+        type: string
+      github_token:
+        description: "if provided an user GH's token, it will push update; requires `push` event"
+        default: ""
+        required: false
+        type: string
```

Comment:
```suggestion
      github_token:
        description: "if provided an user GH's token, it will push update; requires `push` event"
        default: ""
        required: false
        type: string
      use_cashe:
        description: "enable using GH caching for performance boost"
        default: true
        required: false
        type: boolean
```